robot/packages/cloud_package_manager.go

```diff
@@ -299,23 +323,33 @@ func sanitizeURLForLogs(u string) string {
 	return parsed.String()
 }
 
-func (m *cloudManager) loadFile(ctx context.Context, url string, p config.PackageConfig) error {
+func (m *cloudManager) downloadPackage(ctx context.Context, url string, p config.PackageConfig) error {
 	// TODO(): validate integrity of directory.
-	if dirExists(m.localDataPath(p)) {
+	if dirExists(p.LocalDataDirectory(m.packagesDir)) {
```

Comment:
This doesn't feel like a strong enough check for "already downloaded." I see the TODO, but even without full validation, I think it's far too easy to successfully make the directory, but fail at some later step of the process. Maybe a touchfile in the directory that's ".download.complete" or something for now, until the full Validation TODO is addressed.

Let me know if I'm missing some other logic that would allow this to retry here if it failed, and I'm just not realizing it.

Reply:
Talk offline -- still an issue but due to rename at the end, not a urgent one.